src/BreakfastSerializer/Serializer.php

```diff
@@ -63,6 +70,55 @@ public function getDepth()
     /**
      * @inheritdoc
      */
+    public function incrementCurrentDepth()
+    {
+        $this->currentDepth++;
+
+        return $this;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function decrementCurrentDepth()
+    {
+        $this->currentDepth =
+            (1 <= $this->currentDepth) ? 1 : $this->currentDepth = 1;
+
+        return $this;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getCurrentDepth()
+    {
+        return $this->currentDepth;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function resetCurrentDepth()
+    {
+        $this->currentDepth = 1;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function isWithinBounds()
+    {
+        $isValid = (self::MAX_DEPTH_NOT_SET === $this->maxDepth)
+            ? true : ($this->currentDepth <= $this->maxDepth)
+                ? true : false;
```

Comment:
Why yes it is

A thing of beauty is it not :)

On Fri, Apr 24, 2015 at 1:28 PM, Eddie Abou-Jaoude <notifications@github.com
> wrote:

> In src/BreakfastSerializer/Serializer.php
> <https://github.com/brettminnie/breakfast-serializer/pull/13#discussion_r29044187>
> :
>
> > +    /**
> > +     * @inheritdoc
> > +     */
> > +    public function resetCurrentDepth()
> > +    {
> > +        $this->currentDepth = 1;
> > +    }
> > +
> > +    /**
> > +     * @inheritdoc
> > +     */
> > +    public function isWithinBounds()
> > +    {
> > +        $isValid = (self::MAX_DEPTH_NOT_SET === $this->maxDepth)
> > +            ? true : ($this->currentDepth <= $this->maxDepth)
> > +                ? true : false;
>
> nested ternary?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/brettminnie/breakfast-serializer/pull/13/files#r29044187>
> .
>